src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,397 @@
+                            ConstantResult.Left when op.LeftOperand.Syntax is TExpressionSyntax left
+                                => new AnalyzedPattern.Constant(left),
+                            ConstantResult.Right when op.RightOperand.Syntax is TExpressionSyntax right
+                                => new AnalyzedPattern.Constant(right),
+                            _ => null
+                        };
+
+                    case IBinaryOperation op when SupportsRelationalPattern && IsComparisonOperator(op.OperatorKind):
+                        return DetermineConstant(op) switch
+                        {
+                            ConstantResult.Left when op.LeftOperand.Syntax is TExpressionSyntax left
+                                => new AnalyzedPattern.Relational(Negate(op.OperatorKind), left),
+                            ConstantResult.Right when op.RightOperand.Syntax is TExpressionSyntax right
+                                => new AnalyzedPattern.Relational(op.OperatorKind, right),
+                            _ => null
+                        };
+
+                    case IIsTypeOperation op
+                        when SupportsTypePattern && CheckTargetExpression(op.ValueOperand) && op.Syntax is TIsExpressionSyntax node:
+                        return new AnalyzedPattern.Type(node);
+
+                    case IIsPatternOperation op
+                        when SupportsSourcePattern && CheckTargetExpression(op.Value) && op.Pattern.Syntax is TPatternSyntax pattern:
+                        return new AnalyzedPattern.Source(pattern);
+
+                    case IParenthesizedOperation op:
+                        return ParsePattern(op.Operand, guards);
+                }
+
+                return null;
+            }
+
+            private enum BoundKind
+            {
+                None,
+                Lower,
+                Higher,
+            }
+
+            private (SyntaxNode Lower, SyntaxNode Higher) GetRangeBounds(IBinaryOperation op)
+            {
+                if (!(op is { LeftOperand: IBinaryOperation left, RightOperand: IBinaryOperation right }))
+                {
+                    return default;
+                }
+
+                return (GetRangeBound(left), GetRangeBound(right)) switch
+                {
+                    ({ Kind: BoundKind.Lower } low, { Kind: BoundKind.Higher } high)
+                        when CheckTargetExpression(low.Expression, high.Expression) => (low.Value.Syntax, high.Value.Syntax),
+                    ({ Kind: BoundKind.Higher } high, { Kind: BoundKind.Lower } low)
+                        when CheckTargetExpression(low.Expression, high.Expression) => (low.Value.Syntax, high.Value.Syntax),
+                    _ => default
+                };
+
+                bool CheckTargetExpression(IOperation left, IOperation right)
+                    => _syntaxFacts.AreEquivalent(left.Syntax, right.Syntax) && this.CheckTargetExpression(left);
```

Comment:
really confusing to call the local and instance method the same thin fwiw.

Reply:
couldn't think of any other name, it's just a [scoped] overload of the same method